astropy/visualization/wcsaxes/core.py

```diff
@@ -27,6 +28,18 @@
 IDENTITY.wcs.cdelt = [1., 1.]
 
 
+class _WCSAxesArtist(Artist):
+    """This is a dummy artist to enforce the correct z-order of axis ticks,
+    tick labels, and gridlines.
+
+    FIXME: This is a bit of a hack, and it would not be necessary if the tick
+    ticks, tick labels, and gridlines were isntances of Artist subclasses as
+    they are for ordinary Matplotlib axes."""
```

Comment:
I don't understand this comment, as e.g. ticks are an instance of ``Line2D`` which can have a zorder, tick labels are instances of ``Text``, and gridlines are a LineCollection, which are all artists. Could you clarify?

Reply:
`Axes.draw` sorts the artists by zorder and then renders them in sequence. The axes, ticks, tick labels, and gridlines are included in these artists. However, `WCSAxes` disables the native Matplotlib axes ticks, labels, and gridlines. Instead, `WCSAxes.draw` renders ersatz ticks, labels, and gridlines by calling the functions `CoordinateHelper._draw_ticks`, `CoordinateHelper._draw_grid`, etc.

Reply:
Ah right, but they are still actual artists, just not part of the 'real' axes - the comment made it sound as though they weren't artists. Do you mean that because we call draw manually, we don't respect zorder?

Reply:
That's right.

Reply:
@lpsinger - ok cool, could you reword the comment?

Reply:
Done. Take a look again, tell me what you think.